File: Php/bench.php
<?php
/*
 * 	Bench.c
 *
 * 	Sources are under Licence LGPL
 *
 *	Made by Delfosse Jérôme for test purposee. dje@perso.be
 *	18/02/11
 */

$bla='CeciestText';
$myfile='php.txt';

$fh=fopen($myfile,'w')
 or die("can't open file");


$d=65;
$e=65;
$f=65;

$time_start = microtime(true);

for ($a=65; $a<91; $a++)
{
	for ($b=65; $b<91; $b++)
	{
		for ($c=65; $c<91; $c++)
		{
			for ($d=65; $d<91; $d++)
			{
				for ($e=65; $e<91; $e++)
				{
//#					for ($f=65; $f<91; $f++)
//#					{
						$data=$bla.chr($a).chr($b).chr($c).chr($d).chr($e).chr($f);
						$digest = sha1($data);
						fwrite($fh,$data." ".hex2bin($digest)."\n");
						//print_r($digest);echo " ".hex2bin($digest)." \n";
//#					}
				}
			}
		}
	}
}
$time_end = microtime(true);
$time = $time_end - $time_start;

echo "Duration Time : $time seconds\n";


fclose($fh);

function hex2bin($str) {
    $bin = "";
    $i = 0;
    do {
        $bin .= chr(hexdec($str{$i}.$str{($i + 1)}));
        $i += 2;
    } while ($i < strlen($str));
    return $bin;
}

?>

File: Perl/bench.pl
#!/usr/bin/perl
use Time::HiRes qw(gettimeofday tv_interval);
use Digest::SHA1  qw(sha1 sha1_hex sha1_base64);

#	Bench.c
#	Sources are under Licence LGPL
#
#	Made by Delfosse Jérôme for test purposee. dje@perso.be
#	18/02/11


my $bla='CeciestText';
my $data,$digest;
my $elapsed;
my $a,$b,$c,$d,$e,$f;

open (MYFILE, '>perl.txt');
$t0 = [gettimeofday];
$d=65;
$e=65;
$f=65;

for ($a=65; $a<91; $a++)
{
	for ($b=65; $b<91; $b++)
	{
		for ($c=65; $c<91; $c++)
		{
#			for ($d=65; $d<91; $d++)
#			{
#				for ($e=65; $e<91; $e++)
#				{
#					for ($f=65; $f<91; $f++)
#					{
						$data=$bla.chr($a).chr($b).chr($c).chr($d).chr($e).chr($f);
						$digest = sha1($data);
						print MYFILE "$data $digest\n";
#					}
#				}
#			}
		}
	}
}

$elapsed = tv_interval($t0);
$elapsed*=1000;

close (MYFILE); 

print "Time ms: $elapsed \n";

